### ISSUE TITLE
slidesPerGroupAuto skips slide on previous navigation

### ISSUE THREAD

**Issue by aaronadamsCA** [nolimits4web/swiper#4964]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://codesandbox.io/s/swiper-slides-per-view-auto-react-forked-kd3sr?file=/src/App.jsx

### Bug description

When using the (excellent!) new `slidesPerGroupAuto` feature + partially visible slides, forward navigation works as expected, but backward navigation "skips" over a partially visible slide.

Steps to reproduce:

1. Open reproduction link.
2. Click next button.
3. Click next button.
4. Click previous button.

### Expected Behavior

1. Slides 1, 2, 3 fully visible.
2. Slides 4, 5, 6 fully visible.
3. Slides 7, 8, 9 fully visible.
4. Slides 4, 5, 6 fully visible.

### Actual Behavior

4. Slides 3, 4, 5 fully visible. Slide 6 is essentially "skipped".

I would expect backward navigation to work the same as forward navigation, so that each slide has an opportunity to be fully visible in the swiper.

### Swiper version

7.0.5

### Platform/Target and Browser Versions

Tested in Edge (latest) on Windows, and Chrome (latest) on Mac OS.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by bartekjacak** [comment]
I'm struggling with the same issue. Were you able to resolve it?

**Comment by bartekjacak** [comment]
Seems like it's a calculation error occurring when the left edge slide is only partially visible. Temporarily, I've addressed this issue by making sure that there's no partially visible slide on the left edge before my code calls `.slidePrev()`.

How? I've created custom Prev Button (instead of using navigation's `prevEl`) and I manually provide `onClick` function doing this:
```js
if (swiper.isEnd) {
  swiper.slideTo(swiper.snapIndex - 1);
} else {
  swiper.slidePrev();
}
```
